src/http/headers.cr

```diff
@@ -338,6 +359,12 @@ struct HTTP::Headers
     end
   end
 
+  getter? serialized
+
+  private def check_already_serialized
+    raise IO::Error.new("Cannot modify headers after sending") if @serialized
```

Comment:
Shouldn't this use a dedicated error class?

Reply:
The error class was proposed by @straight-shoota. I would also say, that no special error class is required, because this fix only tries to avoid incorrect usage of the API. This exception will never be caught. If this exception occurs, the developer will rewrite his code.